src/hooks/useOAuthFlow.tsx

```diff
@@ -59,7 +59,9 @@ export const OAuthContextProvider = ({
       const { onSuccess, onFail } = params;
       if (!isLoggedIn || !authResult?.refreshToken) {
         await clearAuthResult();
-        onSuccess();
+        if (onSuccess) {
+          onSuccess();
+        }
```

Comment:
Bit of a nit but I would prefer `onSuccess?.()` since we don't care if execution succeeds and it reduces the vertical space and branching flow when reading.

Reply:
Love it.  I want to option-alize all of these `on*` methods - I'll get that in an all-in-one PR here in a sec